src/isomorphic/classic/class/ReactClass.js

```diff
@@ -495,6 +495,17 @@ function mixSpecIntoComponent(Constructor, spec) {
       if (shouldAutoBind) {
         autoBindPairs.push(name, property);
         proto[name] = property;
+        if (__DEV__) {
+          var displayName;
```

Comment:
Can you please move this declaration right after `shouldAutoBind`? It seems a little fishy to declare a local variable inside a `__DEV__` block inside `if` and then use it in other `if` branches outside that particular `__DEV__` block. Uglify will remove it from production builds anyway.